### PR TITLE
Docs: add incipit for options manpage presenting common commands

### PR DESCRIPTION
**Pull request by Delapouite**
Hi

Here's a small but comprehensive intro about options, following the existing ones in [hooks](https://github.com/mawww/kakoune/blob/master/doc/manpages/hooks.asciidoc) and [mapping](https://github.com/mawww/kakoune/blob/master/doc/manpages/mapping.asciidoc) man pages.

### CODE REVIEW

File: doc/manpages/options.asciidoc [hunk 1]
```diff
@@ -46,9 +77,6 @@ Types
 	a set of flags, taking a combination of the given values joined by a
 	'|' character
 
-Refer to the 'scopes' documentation page for more information about
```
Comment (lenormf) [inline doc/manpages/options.asciidoc]: Why was this removed?

Comment (Delapouite) [inline doc/manpages/options.asciidoc]: This line was moved a bit further up.

File: doc/manpages/options.asciidoc [hunk 2]
```diff
@@ -5,6 +5,37 @@ NAME
...
+declare-option [-hidden] <type> <name> [value]
+set-option [-add] <scope> <name> <value>
+update-option <scope> <name>
+unset-option <scope> <name>
+----------------------------------------------
+
+Refer to the 'scopes' documentation page for more information about
+scopes.
+
+To get the current value of an option:
```
Comment (lenormf) [inline doc/manpages/options.asciidoc]: This bit duplicates the contents of the `expansions` page, which increases the probability of having discrepancies in the documentation when either page changes.

#1470 did the same thing as well, if pages' topics intersect then we should redirect one to the other with a "for more information about XXX, refer to the YYY page" type message, instead of potentially creating more opportunities for desynchronization.

Comment (Delapouite) [inline doc/manpages/options.asciidoc]: Yes, I see what you mean.
It puts more burden on our shoulders as doc maintainers with the benefits for kakoune newcomers to have the info right where they need it.

What mechanism could be used to let the user easily jump from one topic to another like hyperlink? vim's `:help` tag system is very nice in this regard.

Comment (lenormf) [inline doc/manpages/options.asciidoc]: You mean having e.g. `|expansions|` in the document, and have the page be opened automatically when hitting `<ret>` (or clicking?) on it? It would be nice, also we could use a character other than `|`, if those sequences aren't interpreted by `a2x`.

**Comment by Delapouite** [comment]
Similar incipit has been added.
https://github.com/mawww/kakoune/blob/master/doc/pages/options.asciidoc
Closing
